app/celer-g4/TrackingAction.cc

```diff
@@ -50,19 +50,23 @@ void TrackingAction::PreUserTrackingAction(G4Track const* track)
                  == !SharedParams::CeleritasDisabled());
     CELER_EXPECT(static_cast<bool>(*params_) == static_cast<bool>(*transport_));
 
-    if (SharedParams::CeleritasDisabled())
-        return;
-
     auto const& allowed_particles = params_->OffloadParticles();
     if (std::find(std::begin(allowed_particles),
                   std::end(allowed_particles),
                   track->GetDefinition())
         != std::end(allowed_particles))
     {
-        // Celeritas is transporting this track
-        ExceptionConverter call_g4exception{"celer0003", params_.get()};
-        CELER_TRY_HANDLE(transport_->Push(*track), call_g4exception);
-        const_cast<G4Track*>(track)->SetTrackStatus(fStopAndKill);
+        if (!SharedParams::CeleritasDisabled())
+        {
+            // Celeritas is transporting this track
+            ExceptionConverter call_g4exception{"celer0003", params_.get()};
+            CELER_TRY_HANDLE(transport_->Push(*track), call_g4exception);
+            const_cast<G4Track*>(track)->SetTrackStatus(fStopAndKill);
+        }
+        else if (SharedParams::KillOffloadTracks())
+        {
+            const_cast<G4Track*>(track)->SetTrackStatus(fStopAndKill);
+        }
```

Comment:
The search through particles (in the no-celeritas, no-kill case, i.e. our geant benchmark case) probably isn't anything to worry about performance wise, but I wonder if it'd be better to have the function look like:
```c++
bool const use_celeritas = !SharedParams::CeleritasDisabled();
bool const kill_em = SharedParams::KillOffloadTracks();

if (!use_celeritas && !kill_em)
    return;

// Now search
if (is_em(track)) {
if (use_celeritas) {
  offload(track);
}
kill(track);
}
```
so avoid the search in the geant4-only cacse, and remove the duplicate stop-and-kill call

Reply:
`kill 'em`

Reply:
🤖 🔪